stats/export.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stats
+
+import "sync"
+
+var (
+	exportersMu sync.Mutex // guards exporters
+	exporters   = make(map[string]Exporter)
+)
+
+// Exporter exports the collected records as view data.
+type Exporter interface {
+	Export(viewData *ViewData)
+}
+
+// RegisterExporter registers an exporter.
+func RegisterExporter(name string, e Exporter) {
+	exportersMu.Lock()
+	defer exportersMu.Unlock()
+
+	exporters[name] = e
```

Comment:
Do you need to delete the previous value if already registered an exporter with that name? Maybe return an error?

Reply:
Not sure if returning an error is helpful. How do the user handle that error?

Most registration patterns in the stdlib (e.g. [sql.Register](https://golang.org/pkg/database/sql/#Register) is panic'ing so the user can detect the problem during the develop time. Do we expect users to register exporters dynamically? If not, I am suggesting to panic. If not, we can replace and document it accordingly.